velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -436,6 +437,70 @@ TEST_F(ArithmeticTest, isNanDouble) {
   EXPECT_EQ(false, isNan(std::nullopt));
 }
 
+TEST_F(ArithmeticTest, normalizeNanFloat) {
+  const auto normalizeNan = [&](std::optional<float> a) {
+    return evaluateOnce<float>("normalize_nan(c0)", a);
+  };
+
+  const auto hasher = [&](std::optional<float> f) {
+    return folly::hasher<float>()(f.value());
+  };
+
+  const auto toInt64 = [&](std::optional<float> f) -> int64_t {
```

Comment:
Is this equivalent to casting (int64_t) ?

This may fail with UndefinedBehaviorSanitizer: float-cast-overflow. You may need to use reinterpret_cast.